feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go

```diff
@@ -0,0 +1,393 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package isis_interface_passive_test
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+const (
+	plenIPv4       = 30
+	plenIPv6       = 126
+	ISISInstance   = "DEFAULT"
+	dutAreaAddress = "49.0001"
+	ateAreaAddress = "49.0002"
+	dutSysID       = "1920.0000.2001"
+	v4Metric       = 100
+	v6Metric       = 100
+	password       = "google"
+)
+
+var (
+	dutPort1Attr = attrs.Attributes{
+		Desc:    "DUT to ATE link1 ",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	atePort1attr = attrs.Attributes{
+		Name:    "ATE to DUT port1 ",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+)
+
+// configureDUT configures all the interfaces on the DUT.
+func configureDUT(t *testing.T) {
+	t.Helper()
+	dc := gnmi.OC()
+	dut := ondatra.DUT(t, "dut")
+	i1 := dutPort1Attr.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+
+	t.Log("Pushing interface config on DUT")
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+	}
+}
+
+// configureIsisDut configures isis configs on DUT.
+func configureIsisDut(t *testing.T, dut *ondatra.DUTDevice, intfName string, dutAreaAddress, dutSysID string) {
```

Comment:
go style says "Isis" and "Dut" must both be "ISIS" and "DUT". So technically this must be named configureISISDUT, but maybe just shorten to configureISIS here.